src/nft.vy

```diff
@@ -567,6 +573,23 @@ def safe_mint(owner: address, uri: String[432]):
     self._safe_mint(owner, token_id, b"")
     self._set_token_uri(token_id, uri)
 
+@external
+def mint_and_start_auction(uri: String[432], patron: address):
+    """
+    @notice Only authorized minters can access this function.
+    @param uri The maximum 432-character user-readable string URI for computing `tokenURI`.
+    @param patron The address of the patron starting the auction.
+    """
+    assert self.is_minter[msg.sender], "AccessControl: access is denied"
```

Comment:
why access is denied? specify reason: sender is not a minter

Reply:
ack: fixed.